### PR TITLE
Relics base scripts and damage refactoring.

### PR DESCRIPTION
**Pull request by FTWRahul**

### CODE REVIEW

File: TakeOne/Assets/_Project/Scripts/Dice/DiceRoller.cs [hunk 1]
```diff
@@ -17,18 +19,18 @@ public class DiceRoller : MonoBehaviour
         [SerializeField] private Vector2 diceForce;
         [SerializeField] private Vector2 diceTorque;
         [SerializeField] private float spawnRadius = 1f;
-        [SerializeField] private DiceManager diceMan;
         
-        public void RollDie(HeroSO diceOwner, DiceSO diceSo ,GameObject dieType = null)
+        public DiceController RollDie(IDiceOwner diceOwner, DiceSO diceSo)
         {
-            if (dieType == null) dieType = defaultDicePrefab;
+            var diePrefab = diceSo.DicePrefab ? diceSo.DicePrefab : defaultDicePrefab;
```
Comment (GitWillG) [inline TakeOne/Assets/_Project/Scripts/Dice/DiceRoller.cs:25]: explain this to me when u can

File: TakeOne/Assets/_Project/Scripts/Relics/RelicSO.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+using UnityEngine;
+
+namespace DiceGame.Relics
+{
+    [CreateAssetMenu(menuName = "Create RelicSO", fileName = "RelicSO", order = 0)]
+    public class RelicSO : ScriptableObject
```
Comment (GitWillG) [inline TakeOne/Assets/_Project/Scripts/Relics/RelicSO.cs:6]: There's more going into the RelicSO right?

File: TakeOne/Assets/_Project/Scripts/Utility/GameEventListenerInterfaces.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+namespace DiceGame.Utility
+{
+    public interface IEncounterEventListener
+    {
+        public void OnEncounterStart();
+        public void OnEncounterEnd();
+    }
+
+    public interface ICombatEventListener
+    {
```
Comment (GitWillG) [inline TakeOne/Assets/_Project/Scripts/Utility/GameEventListenerInterfaces.cs:10]: OnDiceSelected?

**Review by GitWillG — APPROVED** [review]

File: TakeOne/Assets/_Project/Scripts/Relics/RelicSO.cs (hunk 2 above)
Comment (FTWRahul) [reply TakeOne/Assets/_Project/Scripts/Relics/RelicSO.cs:6]: yes

**Pull request merged**